### PR TITLE
Navbar and Hero Banner Created

### PR DESCRIPTION
**Pull request by arkojeet-bera**
The components are not responsive as of now, but they 'll be. Also theming is working but dark colour designs for individual components are not decided yet!
> Several packages  are added for Shad Cn and Acerternity UI.

### CODE REVIEW

File: client/src/components/theme-provider.tsx
Comment (Bismay5467) [inline client/src/components/theme-provider.tsx:1]: Create separate files under "type" folder for declaring global types and interfaces.
Create separate files under "context" folder for declaring different contexts
Create separate files under "hooks" folder for declaring different hooks

File: client/src/pages/Home/HeroBanner.tsx [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+  bgColor: string;
+  textColor: string;
+  borderRadius: string;
+  top?: string;
+  bottom?: string;
+  left?: string;
+  right?: string;
+}
+
+const ChatBox: React.FC<chatBoxComp> = ({
```
Comment (Bismay5467) [inline client/src/pages/Home/HeroBanner.tsx]: Try to use functional declarations for components and functional expressions for utility functions - good coding practices :)

File: client/src/pages/Home/HeroBanner.tsx [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+    <div
+      className="rounded-lg p-4 absolute drop-shadow-xl"
+      style={styleAttributes}
+    >
+      {text}
+    </div>
+  );
+};
+
+const HeroBanner = () => {
```
Comment (Bismay5467) [inline client/src/pages/Home/HeroBanner.tsx]: Single File Single Responsibility
Also try to separate logic from rendering to adhere to coding practices.

**Review by Bismay5467 — CHANGES_REQUESTED** [review]

File: client/src/components/theme-provider.tsx
Comment (ghost) [reply client/src/components/theme-provider.tsx:1]: @Bismay5467 I've created the folders and refactored the code as per suggestions. Please have a look!

File: client/src/pages/Home/HeroBanner/HeroBanner.tsx [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+            </Button>
+          </div>
+        </div>
+        <div className="w-[50%] flex flex-col relative justify-center">
+          <img
+            src={heroImg}
+            alt="Hero Image"
+            className="rounded-tl-[80px] rounded-br-[80px]"
+          />
+          <ChatBox
```
Comment (Bismay5467) [inline client/src/pages/Home/HeroBanner/HeroBanner.tsx]: Avoid rendering same components multiple times. Grab all props in an array (array of props) and just map over that array.

File: client/src/pages/Home/HeroBanner/ChatBox.tsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  bgColor: string;
+  textColor: string;
+  borderRadius: string;
+  top?: string;
+  bottom?: string;
+  left?: string;
+  right?: string;
+}) {
+  const styleAttributes: { [key: string]: string } = {};
+  if (top) styleAttributes.top = top;
```
Comment (Bismay5467) [inline client/src/pages/Home/HeroBanner/ChatBox.tsx]: May refactor the code to something similar to this :

const styleAttributes = {
  ...(top && { top }),
  ...(bottom && { bottom }),
  ...(left && { left }),
  ...(right && { right }),
  backgroundColor: bgColor,
  color: textColor,
  'border-radius': borderRadius
};

File: client/src/hooks/theme-provider.tsx [hunk 5]
```diff
@@ -0,0 +1,73 @@
+import React, { createContext, useContext, useEffect, useState } from 'react';
```
Comment (Bismay5467) [inline client/src/hooks/theme-provider.tsx:1]: This file may later needs refactoring but for now LGTM!

**Review by Bismay5467 — CHANGES_REQUESTED** [review]
@simply-arko left some minor nits, nothing major. Once we get those cleaned up, we can get this PR merged in!!

**Comment by ghost** [comment]
@Bismay5467 I've refactored the codebase as suggested. Please have a look!

**Review by Bismay5467 — APPROVED** [review]

**Pull request merged**